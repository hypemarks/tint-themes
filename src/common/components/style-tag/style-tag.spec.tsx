import { render } from '@testing-library/react';
import React from 'react';
import Style from './style-tag';

describe('Test <Style /> Component', () => {
  it('should render snapshot', () => {
    const customCss = '.post{background:red;}';
    const wrapper = render(<Style customCSS={customCss} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
  it('should render style html tag', () => {
    const customCss = '.post{background:red;}';
    const wrapper = render(<Style customCSS={customCss} />);
    const style = wrapper.queryByTestId('style-tag');
    expect(style).toBeInTheDocument();
    expect(style).toHaveTextContent(customCss);
  });

  it('should not render style html tag', () => {
    const wrapper = render(<Style />);
    const style = wrapper.queryByTestId('style-tag');
    expect(style).not.toBeInTheDocument();
  });
});
