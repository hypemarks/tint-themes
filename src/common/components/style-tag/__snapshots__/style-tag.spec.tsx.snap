// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Style /> Component should render snapshot 1`] = `
<DocumentFragment>
  <style
    data-testid="style-tag"
  >
    .post{background:red;}
  </style>
</DocumentFragment>
`;
